File: 1/part2.lua
---------------
-- Variables --
---------------

-- Input the file into a table
local file = io.open("input.txt", "r");
local arr = {}
for line in file:lines() do
   table.insert (arr, line);
end

local sMilli = os.clock()

---------------
-- Functions --
---------------

function summed2020(x,y,z)
    if x + y + z == 2020 then
        return true
    end
    return false
end

for i, x in pairs(arr) do
    x = tonumber(x)

    for i, y in pairs(arr) do
        y = tonumber(y)
        for i, z in pairs(arr) do
            z = tonumber(z)
            if summed2020(x,y,z) == true then
                local eMilli = os.clock()
                print("The answer is: "..x*y*z..". Duration: "..eMilli - sMilli)
                os.exit()
            end
        end
    end
end

File: 1/part2.rb
#!/usr/bin/env ruby
require 'date'

# Variables
input = IO.readlines("input.txt")

# Functions
def summed2020?(x,y,z)
    return true if x + y + z == 2020
    return false
end

#epochStart = DateTime.now.strftime('%Q').to_i
epochStart = Time.now.to_f 
input.each do |e|
    e = e.to_i
    input.each do |x|
        x = x.to_i
        input.each do |z|
            z = z.to_i
            if summed2020?(e,x,z)
                #epochEnd = DateTime.now.strftime('%Q').to_i
                epochEnd = Time.now.to_f 
                duration = epochEnd - epochStart
                printf "The answer: #{e * x * z}. Duration: #{duration} ms\n"
                exit
            end
        end
    end
end
